File: yunlan-video-server/src/main/java/com/stonewu/fusion/entity/ai/UserApiKey.java
package com.stonewu.fusion.entity.ai;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.stonewu.fusion.common.BaseEntity;
import lombok.*;

/**
 * 用户自带模型密钥实体
 * <p>
 * 对应数据库表：afv_user_api_key
 * <p>
 * 与 afv_api_config（渠道配置）物理隔离：
 * 渠道/地址/协议/代理统一由后台全局维护，本表<b>只存用户自己的密钥</b>，
 * 从根本上避免用户密钥与后台密钥混用、或克隆渠道时误复制后台密钥。
 */
@TableName("afv_user_api_key")
@Data
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserApiKey extends BaseEntity {

    /** 主键ID，自增 */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 归属用户ID（非空，永远属于某个具体用户） */
    private Long userId;

    /** 平台标识：newapi / comfyui（与 ApiConfigService.ALLOWED_PLATFORMS 保持一致） */
    private String platform;

    /** 加密后的 API 密钥 */
    private String apiKey;

    /** 应用ID（可选，非敏感信息） */
    private String appId;

    /** 加密后的应用密钥（可选） */
    private String appSecret;

    /** 状态：0-禁用 1-启用 */
    @Builder.Default
    private Integer status = 1;

    /** 备注说明 */
    private String remark;
}

File: yunlan-video-webui/lib/api/user-api-key.ts
import { http } from "./client";

/**
 * 「我的密钥」—— 普通用户唯一可写的模型相关配置。
 * 与后台渠道（afv_api_config）物理隔离，只存用户自己的密钥，按用户强绑定。
 */

/** 用户自带密钥（列表返回时密钥已脱敏） */
export interface UserApiKey {
  id: number;
  userId: number;
  platform: string;
  apiKey: string;
  appId: string | null;
  appSecret: string | null;
  status: number;
  remark: string | null;
  createTime?: string;
  updateTime?: string;
}

/** 保存我的密钥请求 */
export interface MyApiKeySaveReq {
  platform: string;
  apiKey: string;
  appId?: string;
  appSecret?: string;
}

export const myApiKeyApi = {
  /** 我的密钥列表（密钥已脱敏） */
  list: () => http.get<never, UserApiKey[]>("/api/ai/my-api-key/list"),

  /** 保存我的密钥（按平台覆盖更新） */
  save: (data: MyApiKeySaveReq) =>
    http.post<never, boolean>("/api/ai/my-api-key/save", data),

  /** 删除我的密钥 */
  delete: (id: number) =>
    http.delete<never, boolean>(`/api/ai/my-api-key/delete?id=${id}`),
};
